staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -413,26 +419,23 @@ func (c *AvailableConditionController) processNextWorkItem() bool {
 	return true
 }
 
-func (c *AvailableConditionController) enqueue(obj *apiregistrationv1.APIService) {
-	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
-	if err != nil {
-		klog.Errorf("Couldn't get key for object %#v: %v", obj, err)
-		return
-	}
-
-	c.queue.Add(key)
-}
-
 func (c *AvailableConditionController) addAPIService(obj interface{}) {
 	castObj := obj.(*apiregistrationv1.APIService)
 	klog.V(4).Infof("Adding %s", castObj.Name)
-	c.enqueue(castObj)
+	if castObj.Spec.Service != nil {
+		c.rebuildAPIServiceCache()
```

Comment:
explain why this is done synchronously on this thread.

Reply:
It was slow moving and guaranteed the cache was up to date before ensuring the apiservice for checking